hack/tools/go.mod

```diff
@@ -2,24 +2,25 @@ module sigs.k8s.io/cluster-api-provider-azure/hack/tools
 
 go 1.18
 
-require sigs.k8s.io/cluster-api/hack/tools v0.0.0-20220810165624-90923e703fac
+require (
+	github.com/hashicorp/go-multierror v1.1.1
```

Comment:
Just curious, what's causing us to need to pin this one? Is it temporary and, if so, should we add a comment about conditions for reverting it?

Reply:
It's a package needed to build CAPI's `conversion-verifier`, but `go mod tidy` can't see that and removes it. I can't claim to understand the root importing problem 100%, but I know how it behaves.

We could perhaps revert it by importing a newer version of CAPI's hack/tools, but that appears to have its own set of problems. I did that in https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/d9863531d7bc9373cbdf5ddc879cf21435ecf6b2 but it just ended up creating a problem when checked in to `main`.